### DISCUSSION TITLE
SpanID, TraceId and ParentSpanId in distributed tracing

### DISCUSSION THREAD

**Discussion by jarvisMicroservice** [go-masonry/mortar#63] [Q&A]
Hello,

I am using your greeter service template and have tracing enabled.  The system has two services g-1 and g-2. 

Client calls g-1; g-1 one calls g-2 before a response is sent back to client. 

When I look at logs:

**1. Greeter 1 has** 
app=g-1 parentSpanId=0000000000000000 sampled=true spanId=**34a61a1564b1d277** traceId=**34a61a1564b1d277**

2. **Greeter 2 has**
app=g-2 parentSpanId=70491c8cb3744e72 sampled=true spanId=2b64bab7ba9079d6 traceId=**34a61a1564b1d277**
 
 
 Questions:
 
 1) Why is parentSpandId == 00000 instead of valid value in log 1?  Am I supposed to send some metadata from the client to have it set?
 
 2) is there a way we can make spanId != traceId for first span? (see log 1 where span and trace have same value)
 
 3)  Why is parentspanID of g2 is not equal to spanId of g-1 ?  
 
 
 
 g-1 is calling g-2 using this code
 
 
        ```
wrapper := w.deps.GRPCClientBuilder.Build()
	// Make client and call method
	conn, _ := wrapper.Dial(ctx, "localhost:8080", grpc.WithTransportCredentials(insecure.NewCredentials()))
	c := helloworld.NewGreeterClient(conn)
	c.SayHello(ctx, &helloworld.HelloRequest{Name: "g-2  hello"})
```

**Comment by talgendler** [comment]
You can read more about it here: https://www.jaegertracing.io/docs/1.38/architecture/
